python/tests/integration/arcticdb/test_arctic_additional.py

```diff
@@ -0,0 +1,421 @@
+    verify_dataframe_column(df=result, row_name="NUMBER2", max_type=last_type_c, array=arr_all_c)
+    verify_dataframe_column(df=result, row_name="NUMBER3", max_type=last_type_str, array=arr_all_str)
+    verify_dataframe_column(df=result, row_name="NUMBER5", max_type=last_type_d, array=arr_all_d)
+    verify_dataframe_column(df=result, row_name="NUMBER4", max_type=last_type_bool, array=arr_all_bool)
+
+    assert 3 == len(lib.list_versions(symbol=symbol))
+
+    lib.delete(symbol=symbol,versions=2)
+
+    assert 2 == len(lib.list_versions(symbol=symbol))
+
+    result:pd.DataFrame = lib.read(symbol).data
+    dataframe_dump_to_log("RESULT DF:", result)
+
+    #We want to validate that after delete last version the empty cell are 
+    #filled with 0 not nans
+    verify_dataframe_column(df=result, row_name="NUMBER1", max_type=np.int16, array=concat_all_arrays(arr_b1, arr_b2))
+
+    # As final step we once again repeat staging op with same data
+    # To arrive at final stage
+    lib.write(symbol,df2, staged=True)
+    lib.write(symbol,df3, staged=True)
+    lib.finalize_staged_data(symbol=symbol,mode=StagedDataFinalizeMethod.APPEND)
+
+    result:pd.DataFrame = lib.read(symbol).data
+    dataframe_dump_to_log("RESULT DF:", result)
+    assert 3 == len(lib.list_versions(symbol=symbol))
+
+    #Some final confirmations all is ok
+    verify_dataframe_column(df=result, row_name="NUMBER1", max_type=last_type_b, array=arr_all_b)
+    verify_dataframe_column(df=result, row_name="NUMBER2", max_type=last_type_c, array=arr_all_c)
+
+
+def test_finalize_staged_data_long_scenario(arctic_library_lmdb):
+    """
+        The purpose of of the test is to assure all staged segments along with their data
+        are correctly finalized and resulting
+    """
+
+    lib = arctic_library_lmdb
+    symbol = "symbol"
+
+    cachedDF = CachedDFGenerator(25000, [1])
+
+    total_number_rows: TimestampNumber = TimestampNumber(0, cachedDF.TIME_UNIT) # Synchronize index frequency
+
+    num_rows_initially = 99
+    print(f"Writing to symbol initially {num_rows_initially} rows")
+    df = cachedDF.generate_dataframe_timestamp_indexed(num_rows_initially, total_number_rows, cachedDF.TIME_UNIT)
+    dataframe_dump_to_log("df", df)
+    empty_df = df.drop(df.index)
+    dataframe_dump_to_log("empty", empty_df)
+    total_number_rows = total_number_rows + num_rows_initially
+    lib.write(symbol, data=df, prune_previous_versions=True)
+    cachedParts = CacheParts(df)
+    cachedParts.cache_samples_from(df)
+    chunk_list = generate_chunk_sizes(2, 900, 1100)
+    for chunk_size in chunk_list:
+        df = cachedDF.generate_dataframe_timestamp_indexed(chunk_size, total_number_rows, cachedDF.TIME_UNIT)
+        lib.write(symbol, data=df, validate_index=True, staged=True)
+        ## Unfortunately there is a bug with empty dataframe with index
+        ## Uncomment when this is solved
```

Comment:
An `xfail`'d test with a reference to a bug ticket is more suitable than commented out code

Reply:
yes, indeed that portion is extraxcted in "test_finalize_empty_dataframe". Ideally when fixed it could be added also back to the tests for completeness ... As now we cannot leave it there.